viser/client/src/Titlebar.tsx

```diff
@@ -0,0 +1,176 @@
+import Box from "@mui/material/Box";
+import { Button, Grid, Icon, IconButton, SvgIcon } from "@mui/material";
+import { useContext } from "react";
+import { ViewerContext } from ".";
+
+import * as Icons from '@mui/icons-material';
+type IconName = keyof typeof Icons;
+
+const buttonVariants = ['text', 'contained', 'outlined'];
+function isVariant(variant: unknown): variant is ('text' | 'contained' | 'outlined') {
+    return typeof variant === 'string' && buttonVariants.includes(variant);
+}
+
+interface TitlebarButtonProps {
+    text?: string;
+    href?: string;
+    icon?: string;
+    variant?: string;
+}
+function isButtonData(obj: any): obj is TitlebarButtonProps {
+    return (
+        typeof obj === "object" &&
+        (obj.text === null || typeof obj.text === "string") &&
+        (obj.href === null || typeof obj.href === "string") &&
+        (obj.icon === null || typeof obj.icon === "string") &&
+        (obj.variant === null || typeof obj.variant === "string")
+    );
+}
+
+interface TitlebarImageProps {
+    imageSource: string;
+    alt: string;
+}
+function isImageData(obj: any): obj is TitlebarImageProps {
+    return (
+        typeof obj === "object" &&
+        typeof obj.imageSource === "string" &&
+        typeof obj.alt === "string"
+    );
+}
+
+interface TitlebarPaddingProps {
+    width: string;
+}
+function isPaddingData(obj: any): obj is TitlebarPaddingProps {
+    return (
+        typeof obj === "object" &&
+        typeof obj.width == "string"
+    );
+}
+
+export function TitlebarButton(props: TitlebarButtonProps) {
+    if (props.icon !== null && props.text === null) {
+        return (
+            <IconButton
+            href={props.href ?? ''}>
+                <SvgIcon component={Icons[props.icon as IconName] ?? null} />
```

Comment:
it would be nice to avoid this cast: not sure if it's possible but it would be neat if in Python the type of `icon` was constrained to a Literal of possible icon names and that type somehow made its way into the typescript interface (via `sync_message_defs.py`) and then to here!